packages/devtools_app/lib/src/flutter/screen.dart

```diff
@@ -118,46 +127,61 @@ abstract class ConditionalScreen extends Screen {
   final String conditionalLibrary;
 }
 
-enum DevToolsScreenType {
-  inspector,
-  timeline,
-  memory,
-  performance,
-  logging,
-  info,
-  connect,
-  debugger,
-  network,
-  simple,
-  conditional,
-}
+class DevToolsScreenType {
+  const DevToolsScreenType(this.id, {this.createOverride});
+
+  final String id;
+
+  final Screen Function() createOverride;
+
+  static const inspectorId = 'inspector';
+  static const timelineId = 'timeline';
+  static const memoryId = 'memory';
+  static const performanceId = 'performance';
+  static const networkId = 'network';
+  static const debuggerId = 'debugger';
+  static const loggingId = 'logging';
+  static const infoId = 'info';
+  static const connectId = 'connect';
+  static const simpleId = 'simple';
+
+  static const inspector = DevToolsScreenType(inspectorId);
+  static const timeline = DevToolsScreenType(timelineId);
+  static const memory = DevToolsScreenType(memoryId);
+  static const performance = DevToolsScreenType(performanceId);
+  static const network = DevToolsScreenType(networkId);
+  static const debugger = DevToolsScreenType(debuggerId);
+  static const logging = DevToolsScreenType(loggingId);
+  static const info = DevToolsScreenType(infoId);
+  static const connect = DevToolsScreenType(connectId);
+  static const simple = DevToolsScreenType(simpleId);
 
-// TODO(kenz): we may need to refactor this code if conditional screens need
-// to be created from imports in the same way that non-conditional screens do.
-extension DevToolsScreenTypeExtension on DevToolsScreenType {
   Screen create() {
```

Comment:
lets eliminate the create() method completely. It would be simpler and equally effective to instead filter the list of screens to only include the screen matching the DevToolsScreenId rather than requiring screens to have factories like this.
I'd suggest we remove the DevToolsScreenType class and just use string Ids.

Reply:
Will do some refactoring for this in a follow up PR